src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
```

Comment:
The code can be made more efficient by using an early termination approach when filtering the array. For example, in calculateWomenAverageAge, you can stop searching for children once you find the first one for each woman, instead of searching the entire array every time. This can significantly reduce the time complexity of the function.